GitUI/Avatars/ICachedAvatarProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace GitUI.Avatars
+{
+    /// <summary>
+    /// An <see cref="IAvatarProvider"/> with support for clearing avatar caches.
+    /// </summary>
+    public interface ICachedAvatarProvider : IAvatarProvider
```

Comment:
Managing caching is a separate task not related to getting an avatar.
Suggest to split the interface and use like IAvatarCache or or something that matches the declared.
Explicitly use the cache without hiding the implementation in the class hierarchy.

```
private static ICachedAvatarProvider WrapInCacheAndStaticFallback(IAvatarProvider rootProvider)
{
    var persistentCacheProvider = new AvatarPersistentCache(rootProvider);
    var memoryCachedProvider = new AvatarMemoryCache(persistentCacheProvider);

    var providersWithCache = new ICachedAvatarProvider[]
    {
        persistentCacheProvider,
        memoryCachedProvider,
    };

    var providerWithStaticFallback = new ChainedAvatarProvider(
        memoryCachedProvider,
        UserImageAvatarProvider);

    return new CacheControlAvatarProvider(providerWithStaticFallback, providersWithCache);
}
```